### PR TITLE
Better error message for large elements.

### PR DESCRIPTION
**Pull request by robertwb**
This will cause an exception when the too-large element is emitted, rather than later when the proto is serialized (which happens on another thread and may also cause spurious errors in the data channel consumption).

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/runners/worker/data_plane.py [hunk 1]
```diff
@@ -147,6 +148,10 @@ def maybe_flush(self):
   def flush(self):
     # type: () -> None
     if self._flush_callback:
+      if self.size() > _FLUSH_MAX_SIZE:
+        raise ValueError(
+            f'Buffer size {self.size()} exceeds GRPC limit {_FLUSH_MAX_SIZE}. '
+            'This is likely due to a single element that is too large.')
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/worker/data_plane.py:154]:
```suggestion
            'This is likely due to a single element that is too large. '
            'To mitigate, store large elements in external storage systems, '
            'and use PCollections to pass their metadata, '  
            'or use a custom coder that reduces the element's size.')
```

File: sdks/python/apache_beam/runners/worker/data_plane.py [hunk 2]
```diff
@@ -63,6 +63,7 @@
 
 _DEFAULT_SIZE_FLUSH_THRESHOLD = 10 << 20  # 10MB
 _DEFAULT_TIME_FLUSH_THRESHOLD_MS = 0  # disable time-based flush by default
+_FLUSH_MAX_SIZE = 2 << 30 - 100  # 2GB less some overhead, protobuf/grcp limit
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/worker/data_plane.py:66]:
```suggestion
_FLUSH_MAX_SIZE = (2 << 30) - 100  # 2GB less some overhead, protobuf/grpc limit
```

**Review by tvalentyn — APPROVED** [review]
Thanks!

File: sdks/python/apache_beam/runners/worker/data_plane.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/runners/worker/data_plane.py:66]: Ah, yeah, that - 100 was a last minute addition.

File: sdks/python/apache_beam/runners/worker/data_plane.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/runners/worker/data_plane.py:154]: IMHO generally elements this large are an anti-patter that are likely to cause other problems (e.g. inability to parallelize).

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/worker/data_plane.py]: I see. feel free to correct the suggestion.

**Pull request merged**